src/HealthGPS/energy_balance_model.cpp

```diff
@@ -35,66 +46,163 @@ void EnergyBalanceModel::generate_risk_factors([[maybe_unused]] RuntimeContext &
+    // Update glycogen and water.
+    double k_G = CI_0 / (G_0 * G_0);
+    double G = sqrt(CI / k_G);
+    double W = 2.7 * G;
+
+    // Update extracellular fluid.
+    double Na_b = 4000.0;
+    double Na_f = Na_b * EI / EI_0;
+    double Delta_Na_diet = Na_f - Na_b;
+    double ECF = ECF_0 + (Delta_Na_diet - xi_CI * (1.0 - CI / CI_0)) / xi_Na;
+
+    // Update resting metabolic rate (Mifflin-St Jeor).
+    double RMR = 9.99 * BW_0 + 6.25 * H * 100.0 - 4.92 * person.age;
+    RMR += person.gender == core::Gender::male ? 5.0 : -161.0;
+    RMR *= 4.184; // kcal to kJ
+
+    double delta_0 = ((1.0 - beta_TEF) * PAL - 1.0) * RMR / BW_0;
+
+    // Update thermic effect of food and adaptive thermogenesis.
+    double TEF = beta_TEF * EI - EI_0;
+    double AT = beta_AT * EI - EI_0;
+
+    // TODO: Update intercept value.
+    const double K = 0.0; // TODO: Intercept value.
+
+    // Energy partitioning.
+    const double C = 10.4 * rho_L / rho_F;
+    const double p = C / (C + F_0);
+
+    // First equation ax + by = e.
+    double a1 = p * rho_F;
+    double b1 = -(1.0 - p) * rho_L;
+    double c1 = p * rho_F * F_0 - (1 - p) * rho_L * L_0;
+
+    // Check no baseline adjustment in trial run.
+    if (!final_run)
+        final_shift = 0.0;
+
+    // Second equation cx + dy = f.
+    double a2 = gamma_F + delta_0;
+    double b2 = gamma_L + delta_0;
+    double c2 = EI - final_shift - K - TEF - AT - delta_0 * (G + W + ECF);
+
+    // Update body fat and lean tissue steady state.
+    double steady_F = -(b1 * c2 - b2 * c1) / (a1 * b2 - a2 * b1);
+    double steady_L = -(c1 * a2 - c2 * a1) / (a1 * b2 - a2 * b1);
+
+    double x = p * eta_L / rho_L + (1.0 - p) * eta_F / rho_F;
+    double tau = rho_L * rho_F * (1.0 + x) /
+                 ((gamma_F + delta_0) * (1.0 - p) * rho_L + (gamma_L + delta_0) * p * rho_F);
+
+    // Update body fat and lean tissue.
+    double F = steady_F + (F_0 - steady_F) * exp(-365.0 / tau);
+    double L = steady_L + (L_0 - steady_L) * exp(-365.0 / tau);
+
+    // Update body weight.
+    double BW = F + L + G + W + ECF;
+
+    double delta_BW = delta_0 * BW;
+
+    // Update energy expenditure.
+    double EE =
+        (final_shift + K + gamma_F * F + gamma_L * L + delta_BW + TEF + AT + EI * x) / (1.0 + x);
+
+    if (final_run) { // Final run:
+        // Save state and return nothing.
+        person.risk_factors[H_key] = H;
+        person.risk_factors[BW_key] = BW;
+        person.risk_factors[PAL_key] = PAL;
+        person.risk_factors[F_key] = F;
+        person.risk_factors[L_key] = L;
+        person.risk_factors[ECF_key] = ECF;
+        person.risk_factors[EI_key] = EI;
+        person.risk_factors[EE_key] = EE;
+        return std::array<double, 2>{};
+    } else { // Trial run:
+        // return BW and baseline adjustment coefficient.
+        double trial_adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
+        return std::array<double, 2>{BW, trial_adjust};
+    }
```

Comment:
I'd drop the `else`. Also you don't need to specify the type here because the compiler can infer it:

```suggestion
        return {};
    }

    // Trial run:
    // return BW and baseline adjustment coefficient.
    double trial_adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
    return {BW, trial_adjust};
```